### PR TITLE
Refactor : move current year and week from study to weekly optimization problem

### PR DESCRIPTION
**Pull request by guilpier-code**
The purpose is to move the information of the current year and week from the Study to the weekly optimization problem.
Why ?
- Because weekly problem is more the location of current year and week (the problem knows what year and week it's solving)
- It makes the code cleaner (reduces dependency on the study)
- It makes easier some changes to come (export MPS files when optimization is daily)

Some collateral cleanings were also made.

### CODE REVIEW

File: src/solver/utils/mps_utils.cpp [hunk 1]
```diff
@@ -320,49 +324,60 @@ void OPT_EcrireJeuDeDonneesLineaireAuFormatMPS(void* Prob, int optNumber, uint n
...
+                                           int year,
+                                           int week,
+                                           int optNumber) :
+    I_MPS_writer(year, week, optNumber),
+    solver_(solver)
 {
 }
 void fullOrToolsMPSwriter::runIfNeeded(Solver::IResultWriter::Ptr writer)
 {
     // Make or-tools print the MPS files leads to a crash !
```
Comment (flomnes) [inline src/solver/utils/mps_utils.cpp]: This comment is now irrelevant, please remove it

File: src/solver/utils/mps_utils.h [hunk 2]
```diff
@@ -28,39 +28,52 @@ void OPT_EcrireJeuDeDonneesLineaireAuFormatMPS(void*,
 class I_MPS_writer
 {
 public:
-    I_MPS_writer() = default;
+    I_MPS_writer(int year, int week, int currentOptimNumber) : 
+       year_(year), week_(week), current_optim_number_(currentOptimNumber)
+    {}
+    I_MPS_writer()
+    {}
```
Comment (flomnes) [inline src/solver/utils/mps_utils.h]:
```suggestion
    I_MPS_writer(int year, int week, int currentOptimNumber) : 
       year_(year), week_(week), current_optim_number_(currentOptimNumber) = default;
    I_MPS_writer() = default;
```

File: src/solver/simulation/sim_structure_probleme_economique.h [hunk 3]
```diff
@@ -490,6 +490,10 @@ struct AdequacyPatchParameters
 
 struct PROBLEME_HEBDO
 {
+    int dayInTheYear = 0;
+    int weekInTheYear = 0;
+    int year = 0;
```
Comment (flomnes) [inline src/solver/simulation/sim_structure_probleme_economique.h]: `unsigned int`

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/solver/utils/mps_utils.h (hunk 2 above)
Comment (guilpier-code) [reply src/solver/utils/mps_utils.h]: Having a **default** constructor with an initialization list does not compile : 
```C++
I_MPS_writer(int year, int week, int currentOptimNumber) : 
       year_(year), week_(week), current_optim_number_(currentOptimNumber) = default;
```

File: src/solver/utils/mps_utils.cpp (hunk 1 above)
Comment (guilpier-code) [reply src/solver/utils/mps_utils.cpp]: Done
**Question** : from which version of or-tools was the problem fixed ?

File: src/solver/simulation/sim_structure_probleme_economique.h (hunk 3 above)
Comment (guilpier-code) [reply src/solver/simulation/sim_structure_probleme_economique.h]: Done

Comment (guilpier-code) [reply src/solver/simulation/sim_structure_probleme_economique.h]: done

File: src/solver/utils/mps_utils.cpp (hunk 1 above)
Comment (guilpier-code) [reply src/solver/utils/mps_utils.cpp]: Done

File: src/solver/utils/mps_utils.h (hunk 2 above)
Comment (guilpier-code) [reply src/solver/utils/mps_utils.h]: **default** identifier after initialization list does not compile

File: src/solver/simulation/sim_structure_probleme_economique.h [hunk 4]
```diff
@@ -490,6 +490,10 @@ struct AdequacyPatchParameters
 
 struct PROBLEME_HEBDO
 {
+    uint dayInTheYear = 0;
```
Comment (flomnes) [inline src/solver/simulation/sim_structure_probleme_economique.h:493]: Unused

**Review by flomnes — CHANGES_REQUESTED** [review]

File: src/solver/simulation/sim_structure_probleme_economique.h (hunk 4 above)
Comment (guilpier-code) [reply src/solver/simulation/sim_structure_probleme_economique.h:493]: It is removed in #1041

**Review by flomnes — APPROVED** [review]

**Pull request merged**